source/agora/test/Ledger.d

```diff
@@ -54,15 +54,21 @@ unittest
     foreach (block_idx; 0 .. 10)  // create 10 blocks
     {
         // create enough tx's for a single block
-        auto txs = blocks[block_idx].spendable().map!(txb => txb.sign()).array();
+        Transaction[] txs = blocks[block_idx].spendable().map!(txb => txb.sign()).array;
 
-        // send it to one node
-        txs.each!(tx => node_1.putTransaction(tx));
+        txs.each!((tx)
+        {
+            // send it to one node
+            node_1.putTransaction(tx);
+            // wait for all nodes get the txs
+            nodes.each!(node => node.hasTransactionHash(hashFull(tx)).retryFor(2.seconds));
+        });
```

Comment:
```suggestion
        txs.each!(tx => node_1.putTransaction(tx));
        // wait for all nodes get the txs
        txs.each!(tx =>
            nodes.each!(node => node.hasTransactionHash(hashFull(tx)).retryFor(2.seconds))
        );
```

Instead of sending the tx, then waiting for all nodes, send all txs, then wait for all nodes ?

Reply:
In some other tests that is what we do. Just thought I would mix it up a bit. Although I suppose we test gossiping in another test.

Reply:
Just re-read your comment and realized what you meant. Yes it would be more efficient. Will update.

Reply:
Updated